app/services/finance/ecf/mentor/statement_calculator.rb

```diff
@@ -0,0 +1,113 @@
+# frozen_string_literal: true
+
+module Finance
+  module ECF
+    module Mentor
+      class StatementCalculator
+        class << self
+          def event_types
+            %i[
```

Comment:
Should we couple/delegate to the output calculator here as it will always need to be the same types?

Reply:
yeah I tried that but a class method needs to be there in order to use in the `define_method` loop..

Reply:
You could keep the methods and just avoid redefining the values, so something like:

```
def declaration_types
  OutputCalculator.supported_declaration_types
end
```

Then in the `StatementCalculator`:

```
def self.supported_declaration_types
  DECLARATION_TYPE_FEE_PROPORTIONS.stringify_keys.keys
end
```

I'm assuming that these will always be the same value and it would just make it clear that if you want to add a new declaration type to the statement calculator then the output calculator has to change.

Just an thought, I don't feel that strongly about it.